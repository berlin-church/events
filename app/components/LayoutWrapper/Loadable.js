/**
 *
 * Asynchronously loads the component for LayoutWrapper
 *
 */

import loadable from 'utils/loadable';

export default loadable(() => import('./index'));
