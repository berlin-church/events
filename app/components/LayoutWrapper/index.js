/**
 *
 * LayoutWrapper
 *
 */

import React, { memo } from 'react';

import { makeStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';

const useStyles = makeStyles({
  root: {
    flexGrow: 1,
  },
  appBar: {
    backgroundColor: '#000',
    height: '64',
  },
  title: {
    justifyContent: 'center',
    color: '#fff',
  },
  logo: {
    width: '150px',
  },
});

function LayoutWrapper() {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <AppBar className={classes.appBar} position="static">
        <Toolbar className={classes.title}>
          <img
            className={classes.logo}
            src="//static1.squarespace.com/static/589397498419c2f6476f2c77/t/5b28efa970a6ad0778246fc3/1539028385210/?format=1500w"
            alt="Berlin International Community Church"
          />
        </Toolbar>
      </AppBar>
    </div>
  );
}

export default memo(LayoutWrapper);
