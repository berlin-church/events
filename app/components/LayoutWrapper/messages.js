/*
 * LayoutWrapper Messages
 *
 * This contains all the text for the LayoutWrapper component.
 */

import { defineMessages } from 'react-intl';

export const scope = 'app.components.LayoutWrapper';

export default defineMessages({
  header: {
    id: `${scope}.header`,
    defaultMessage: '',
  },
});
