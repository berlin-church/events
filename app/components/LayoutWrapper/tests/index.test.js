/**
 *
 * Tests for LayoutWrapper
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import LayoutWrapper from '../index';
import { DEFAULT_LOCALE } from '../../../i18n';

describe('<LayoutWrapper />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <LayoutWrapper />
      </IntlProvider>,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <LayoutWrapper />
      </IntlProvider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
