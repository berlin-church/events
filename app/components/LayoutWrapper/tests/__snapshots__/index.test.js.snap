// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LayoutWrapper /> Should render and match the snapshot 1`] = `
<div
  class="makeStyles-root-45"
>
  <header
    class="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary makeStyles-appBar-46"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters makeStyles-title-47"
    >
      <img
        alt="Berlin International Community Church"
        class="makeStyles-logo-48"
        src="//static1.squarespace.com/static/589397498419c2f6476f2c77/t/5b28efa970a6ad0778246fc3/1539028385210/?format=1500w"
      />
    </div>
  </header>
</div>
`;
